### PR TITLE
User signup acceptance test

### LINKED ISSUE #2: User signup/registration
```
As a user
In order to access the application
I need to register an account
```

### PR DESCRIPTION
**Pull request by j0sv**
Fixes #2 User Signup/registration

### Changes proposed by this PR

- Added User Signup/registration test - passed 
- Added User Signup/registration test with missing name, invalid password and password confirmation not matching - passed 

### What did you learn working on this PR?
How to write feature to be able to recycle them in the step definitions.

### CODE REVIEW

File: features/step_definitions/basic_step.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+Then(/^show me the page$/) do
+  save_and_open_page
+end
+
+Given(/^the following user exist$/) do |table|
+  table.hashes.each do | user |
+    User.create(user)
```
Comment (tochman) [inline features/step_definitions/basic_step.rb]: Factory Girl is added to Gemfile. Please use it.

**Review by tochman — CHANGES_REQUESTED** [review]

File: features/step_definitions/basic_step.rb (hunk 1 above)
Comment (j0sv) [reply features/step_definitions/basic_step.rb]: Step definition is removed. New step definition is added to branch for login test.

**Review by tochman — APPROVED** [review]

File: features/user_signup.feature [hunk 2]
```diff
@@ -2,13 +2,27 @@ Feature: As a user
   In order to access the application
   I need to register an accoun
```
Comment (diraulo) [inline features/user_signup.feature]: `account`

File: features/user_signup.feature [hunk 3]
```diff
@@ -2,13 +2,27 @@ Feature: As a user
...
+    And I click "Sign up" button
+    Then I should be on "Sign up" page
+    And I fill in "Email" with "john@doe.com"
+    And I fill in "Password" with "abc123"
+    And I click "Create" button
+    Then I should see the message "Password confirmation doesn't match Password"
+    And I should see the message "Password is too short (minimum is 8 characters)"
+    And I should see the message "Name can't be blank"
+  Scenario: Register new user
+
```
Comment (diraulo) [inline features/user_signup.feature]: remove line 17 and 18

File: features/user_signup.feature [hunk 4]
```diff
@@ -2,13 +2,27 @@ Feature: As a user
...
-  When I visit "landing" page
-  And I click "Sign up"
-  Then I should be on "Sign up" page
-  When I fill in "Name" with "John"
-  And I fill in "Email" with "john@doe.com"
-  And I fill in "Password" with "abcde1234"
-  And I filll in "Password confirmation" with "abcde1234"
-  And I click "Create" button
-  Then I should see the message "Welcome! You have signed up successfully."
+#Sad path
```
Comment (diraulo) [inline features/user_signup.feature]: remove this..

**Review by diraulo — CHANGES_REQUESTED** [review]
Just minor cleanup on the feature file

**Review by diraulo — APPROVED** [review]
LGTM 👍

**Comment by diraulo** [comment]
Fixed #2

**Comment by diraulo** [comment]
Fixes #2

**Pull request merged**